cloud/src/meta-store/versioned_value.h

```diff
@@ -0,0 +1,149 @@
+    VersionedRangeGetIterator& operator=(VersionedRangeGetIterator&& other) noexcept {
+        if (this != &other) {
+            iter_ = std::move(other.iter_);
+            error_code_ = other.error_code_;
+            has_find_ = other.has_find_;
+            current_version_ = other.current_version_;
+            current_key_ = other.current_key_;
+            current_value_ = other.current_value_;
+            snapshot_version_ = other.snapshot_version_;
+        }
+        return *this;
+    }
+
+    bool has_next();
+
+    bool is_valid() const { return error_code_ == TxnErrorCode::TXN_OK && iter_->is_valid(); }
+
+    TxnErrorCode error_code() const {
+        return error_code_ == TxnErrorCode::TXN_OK ? iter_->error_code() : error_code_;
+    }
+
+    std::optional<Element> next();
+
+    std::optional<Element> peek();
+
+private:
+    std::tuple<std::string_view, Versionstamp> parse_key(std::string_view key);
+
+    Versionstamp snapshot_version_;
+
+    TxnErrorCode error_code_ = TxnErrorCode::TXN_OK;
+    bool has_find_ = false; // Whether we have found a valid key in the current iteration.
+    Versionstamp current_version_;
+    std::string_view current_key_;
+    std::string_view current_value_;
+    std::unique_ptr<FullRangeGetIterator> iter_;
+};
+
+// Get a range of versioned value from the transaction.
+//
+// This function performs a range query on versioned documents stored with key format
+// "key_prefix + versionstamp". It returns an iterator that can traverse through
+// all matching documents within the specified key range (in reverse order).
+//
+// It returns a iterator. The iterator returns documents in reverse order (latest first)
+// and automatically filters out documents newer than the snapshot_version.
+std::unique_ptr<VersionedRangeGetIterator> versioned_get_range(
+        Transaction* txn, std::string_view begin, std::string_view end,
+        const VersionedRangeGetOptions& opts);
+
+// Remove a versioned document from the transaction by key prefix and versionstamp.
+void versioned_remove(Transaction* txn, std::string_view key_prefix, Versionstamp v);
+
+// Remove a versioned document from the transaction by key.
+void versioned_remove(Transaction* txn, std::string_view key);
+
+// Put a versioned value into the transaction with a specific versionstamp.
+bool versioned_put(Transaction* txn, std::string_view key_prefix, Versionstamp v,
+                   std::string_view value);
+
+// Put a versioned value, the versionstamp will be generated automatically.
+bool versioned_put(Transaction* txn, std::string_view key_prefix, std::string_view);
+
+// Get a versioned value from the transaction by key prefix and versionstamp.
+//
+// The versionstamp of the document will be returned in `v`.
+TxnErrorCode versioned_get(Transaction* txn, std::string_view key_prefix,
+                           Versionstamp snapshot_version, Versionstamp* v, std::string* value,
+                           bool snapshot = true);
```

Comment:
use snapshot by default is risky,
keep consistent with the strongest isolation when read from kv, read by snapshot accordingly